src/Domain/FunctionsOfTime/FixedSpeedCubic.cpp

```diff
@@ -0,0 +1,88 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/FunctionsOfTime/FixedSpeedCubic.hpp"
+
+#include <cmath>
+#include <memory>
+
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace domain::FunctionsOfTime {
+FixedSpeedCubic::FixedSpeedCubic(const double initial_function_value,
+                                 const double initial_time,
+                                 const double velocity,
+                                 const double decay_timescale) noexcept
+    : initial_function_value_(initial_function_value),
+      initial_time_(initial_time),
+      velocity_(velocity),
+      squared_decay_timescale_(square(decay_timescale)) {}
```

Comment:
I'm not sure if `tau=0` should be forbidden, but it seems like generally bad things will happen if `tau=0` and `t_0=0`. I'm not sure if the check should be in the constructor or if an `ASSERT` in the `func_and_derivs` to check that we aren't dividing by zero would be better. In either case, I think it would be good to have a check to avoid division by zero

Reply:
The division by zero only happens if tau=0 and you evaluate at t=t_0. Since you can't know in advance what points the user wants to evaluate the function at, I figure the right thing to do is to assert when the denominator is zero when evaluating func_and_derivs.